firmware/library/L2_Utilities/util.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <cstdint>
+
+namespace Utility
+{
+static uint32_t Random(uint32_t min, uint32_t max);
+static void SetRandomSeed();
+static uint32_t BitExtract(uint32_t extractee, uint8_t start_bit,
+                           uint8_t width);
+static uint32_t BitPlace(uint32_t placee, uint8_t s_bit, uint8_t bits,
+                         uint8_t width);
+static uint32_t BitPlace(uint32_t placee, uint8_t s_bit, bool bit,
+                         uint8_t width);
+static uint32_t BitSet(uint32_t setee, uint8_t bit);
+static uint32_t BitClear(uint32_t clearee, uint8_t bit);
+static uint32_t BitRead(uint32_t readee, uint8_t bit);
```

Comment:
Would prefer bit_position, since the meaning of this `bit` variable and the above `bool bit` are different. This is to reduce confusion.